imagick/magick_wand_image.go

```diff
@@ -475,19 +475,19 @@ func (mw *MagickWand) CropImageToTiles(cropGeom string) error {
 
 	defer C.MagickDestroyImage(newImg)
 
-	newWand := NewMagickWandFromImage(&Image{newImg})
-	defer newWand.Destroy()
-
-	if err := mw.AddImage(newWand); err != nil {
-		return fmt.Errorf("error replacing image: %v", err)
-	}
-
-	mw.PreviousImage()
-
 	if err := mw.RemoveImage(); err != nil {
 		return fmt.Errorf("error removing current image: %v", err)
 	}
 
+	for next := newImg; next != nil; next = C.GetNextImageInList(next) {
+		tempWand := NewMagickWandFromImage(&Image{next})
+		if err := mw.AddImage(tempWand); err != nil {
+			return fmt.Errorf("error replacing image: %v", err)
+		}
+		tempWand.Destroy()
+	}
+	mw.SetFirstIterator()
```

Comment:
I see the logic here is meant to get all the possible tile images from the crop operation and append them, whereas the previous code only added the first one.
My only concern here is that you remove the current image first, and then try to add the new images. If adding the new images fails, then it leaves the wand with the original removed. The previous code removed the original image after it success in adding the new one. 
I wonder if you could first save the index with `mw.GetIteratorIndex()`, then add the new images, and then at the end you can do `mw.SetIteratorIndex(idx)`, remove the original image, and then call `mw.ResetIterator()` to move it past the appended images.
What do you think?

Reply:
I think I cloud  just append new image first, after all then do mw.ResetIterator() and call mw.RemoveImage(), resulting the mw point to first tile. Thus prevent the origin image removed before add any tile image.

Reply:
Cheers for that update.